Vikta-API-testing-sample/src/test/java/com/test/api/category/CategoryTest.java

```diff
@@ -57,19 +78,234 @@ public void checkCategoryInAllCategoriesList() {
                 .extract()
                 .as(CategoryDTO[].class);
 
-        assertThat("Category wasn't found in all categories list",
+        assertThat(String.format("Category %s wasn't found in all categories list", categoryId),
                 Arrays.asList(categories),
                 hasItem(equalTo(expectedCategory)));
     }
 
+    @Test
+    public void canListAllCategories(){
+        String pathList = getTestEnvironment().getCategoryPath() + "/list";
+
+        CategoryDTO[] categories = given().spec(defaultRequestSpec())
+                .when()
+                .get(getTestEnvironment().getCategoriesPath())
+                .then()
+                .spec(defaultResponseSpec())
+                .extract()
+                .as(CategoryDTO[].class);
+
+        CategoryDTO[] listOfCategories = given().spec(defaultRequestSpec())
+                .when()
+                .get(pathList)
+                .then()
+                .spec(defaultResponseSpec())
+                .extract()
+                .as(CategoryDTO[].class);
+
+        assertThat(listOfCategories).isEqualTo(categories);
+    }
+
+    @Test
+    public void canFindEntity(){
+        final String pathList = getTestEnvironment().getCategoryPath() + "/list";
+        final String searchPath = getTestEnvironment().getCategoryPath() + "/search";
+        final String termQueryParam = "term";
+        final int size = 3;
+        Random random = new Random();
+
+        List <CategoryDTO> listOfCategories = given().spec(defaultRequestSpec())
+                .when()
+                .get(pathList)
+                .then()
+                .spec(defaultResponseSpec())
+                .extract()
+                .body().jsonPath().getList(".", CategoryDTO.class);
+
+        StringBuilder termBuilder = new StringBuilder();
+        for(int i=0; i<size; i++){
+            termBuilder.append(listOfCategories.get(random.nextInt(listOfCategories.size())).getTitle());
```

Comment:
if You'll pick the same category twice - which is possible with random(), the test will fail